packages/react-native/local-cli/generator-macos/templates/macos/HelloWorld.xcodeproj/project.pbxproj

```diff
@@ -347,6 +347,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				CLANG_ENABLE_MODULES = YES;
```

Comment:
I wish all BuildSettings were removed from the pbxproj and put in an xcconfig so it's easier to share for cases like this. I haven't found a simple way to do so in cocoapods. Meanwhile, I guess we'll keep doing this :D